js/scripts/util.js

```diff
@@ -127,4 +141,86 @@ const checkBlockCollisions = (entity) => {
     entity.boundingBox = new BoundingBox(entity.pos, entity.constructor.SCALED_SIZE);
 
     return collisions;
+};
+
+// The following is necessary because we must change the listeners for different modes (right now, gameplay and dialog).
+/** Contains all functions called as event handlers. */
+const EVENT_HANDLERS = {
```

Comment:
I'm not certain this belongs in util, as it seems like it is relevant only to the game engine. I understand game engine is cluttered enough as is but I'm not sure this is the right place.

Reply:
I don't think I agree with you. Realistically, it has little to do with the game engine, it's only that the game engine is what's going to be calling these. These would be in the Canvas class (if we had one) but I think that util is okay. In the html comment above the util import, it says "Scripts containing info/functions for classes" which is exactly what this is.